analysis/exponential.lean

```diff
@@ -127,6 +127,36 @@ by simp [log, lt_irrefl]
 @[simp] lemma log_one : log 1 = 0 :=
 exp_injective $ by rw [exp_log zero_lt_one, exp_zero]
 
+noncomputable definition nth_root (x : ℝ) (n : ℕ) : ℝ := 
+if x = 0 then 0 else exp ((log x) / n)

```

Comment:
It may be nice to have a lemma like
```lean
lemma nth_root_def {x : R} (hx : x \ne 0) {n : N} : nth_root x n = exp (log x / n) :=
dif_neg hx -- or whatever the proof really is
```
to make it easier to manipulate `nth_root` directly without relying on `simp`.